src/Domain/LeanCode.CQRS.Execution/DevResponseLoggerElement.cs

```diff
@@ -0,0 +1,34 @@
+
+using LeanCode.Contracts;
+using LeanCode.Contracts.Security;
+using LeanCode.Pipelines;
+using Microsoft.Extensions.Hosting;
+
+namespace LeanCode.CQRS.Execution
+{
+    public class DevResponseLoggerElement<AppContext, TObj, TRes> : IPipelineElement<AppContext, TObj, TRes>
+        where AppContext : notnull, IPipelineContext
+    {
+        public readonly Serilog.ILogger logger = Serilog.Log.ForContext<DevResponseLoggerElement<AppContext, TObj, TRes>>();
+        public readonly IHostEnvironment environment;
+
+        DevResponseLoggerElement(IHostEnvironment env)
+        {
+            logger = Serilog.Log.ForContext<DevResponseLoggerElement<AppContext, TObj, TRes>>();
+            environment = env;
+        }
+
+        public Task<TRes> ExecuteAsync(
+            AppContext ctx,
+            TObj input,
+            Func<AppContext, TObj, Task<TRes>> next)
+        {
+            if (environment.IsDevelopment())
```

Comment:
```suggestion
            if (!environment.IsProduction())
```
Maybe? I'm not sure - we probably should ask on our internal Slack.